agents/telegraf.go

```diff
@@ -97,27 +91,61 @@ func (tr *TelegrafRunner) Load(agentBasePath string) error {
 	tr.ingestHost = host
 	tr.ingestPort = port
 	tr.basePath = agentBasePath
+	tr.configServerToken = uuid.NewV4().String()
+	tr.configServerHandler = func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("authorization") != "Token " + tr.configServerToken {
+			http.Error(w, "unauthorized", http.StatusUnauthorized)
+			return
+		}
+		_, err = w.Write(tr.tomlWebPage)
+		if err != nil {
+			log.Errorf("Error writing config page %v", err)
+		}
+	}
+
+	serverId := uuid.NewV4().String()
+	tr.configServerMux = http.NewServeMux()
+	tr.configServerMux.Handle("/" + serverId, tr.configServerHandler)
+
+	// Get the next available port
+	listener, err := net.Listen("tcp", ":0")
```

Comment:
This should probably be
```suggestion
	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

to bind only to loopback.